diff -uNr hddtemp-0.2/hddtemp.c hddtemp-0.2.orig/hddtemp.c
--- hddtemp-0.2/hddtemp.c	Wed Jul 17 08:19:58 2002
+++ hddtemp-0.2.orig/hddtemp.c	Wed Jul 17 08:19:25 2002
@@ -27,12 +27,11 @@
 #include <regex.h>
 #include <sys/ioctl.h>
 #include <linux/hdreg.h>
+#include <linux/types.h>
 
 #include "hddtemp.h"
 
-#ifdef ARCH_I386
-typedef unsigned short u16;
-#endif
+typedef __u16 u16;
 
 #define swapb(x) \
 ({ \
Binary files hddtemp-0.2/hddtemp.o and hddtemp-0.2.orig/hddtemp.o differ
